inference-engine/tests/functional/plugin/shared/src/behavior/set_blob_of_kind.cpp

```diff
@@ -0,0 +1,139 @@
+            subBlobs.push_back(makeBlobOfKind(subBlobDesc, BlobKind::Simple));
+        }
+        return make_shared_blob<BatchedBlob>(subBlobs);
+    }
+    default:
+        THROW_IE_EXCEPTION << "Test does not support the blob kind";
+    }
+}
+
+bool isBatchedBlobSupported(const std::shared_ptr<Core>& core, const LayerTestsUtils::TargetDevice& targetDevice) {
+    const std::vector<std::string> supported_metrics = core->GetMetric(targetDevice, METRIC_KEY(SUPPORTED_METRICS));
+
+    if (std::find(supported_metrics.begin(), supported_metrics.end(),
+                 METRIC_KEY(OPTIMIZATION_CAPABILITIES)) == supported_metrics.end()) {
+        return false;
+    }
+
+    const std::vector<std::string> optimization_caps =
+        core->GetMetric(targetDevice, METRIC_KEY(OPTIMIZATION_CAPABILITIES));
+
+    return std::find(optimization_caps.begin(), optimization_caps.end(),
+                    METRIC_VALUE(BATCHED_BLOB)) != optimization_caps.end();
+}
+
+bool isBlobKindSupported(const std::shared_ptr<Core>& core, const LayerTestsUtils::TargetDevice& targetDevice, BlobKind blobKind) {
+    switch (blobKind) {
+    case BlobKind::Simple:
+        return true;
+    case BlobKind::Compound:
+        return false;
+    case BlobKind::BatchOfSimple:
+        return isBatchedBlobSupported(core, targetDevice);
+    default:
+        THROW_IE_EXCEPTION << "Test does not support the blob kind";
+    }
+}
+
+} // namespace
+
+Blob::Ptr SetBlobOfKindTest::GenerateInput(const InferenceEngine::InputInfo& info) const {
+    return makeBlobOfKind(info.getTensorDesc(), blobKind);
+}
+
+void SetBlobOfKindTest::Run() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    LoadNetwork();
+
+    if (isBlobKindSupported(core, targetDevice, blobKind)) {
+        Infer();
+        Validate();
+    } else {
+        ExpectSetBlobThrow();
+    }
+}
+
+void SetBlobOfKindTest::ExpectSetBlobThrow() {
+    inferRequest = executableNetwork.CreateInferRequest();
+
+    for (const auto &input : executableNetwork.GetInputsInfo()) {
+        const auto &info = input.second;
+        auto blob = GenerateInput(*info);
+        EXPECT_THROW(inferRequest.SetBlob(info->name(), blob),
+                     InferenceEngine::details::InferenceEngineException);
+    }
+}
+
+void SetBlobOfKindTest::SetUp() {
+    SizeVector IS{4, 3, 6, 8};
+    std::tie(blobKind, targetDevice, configuration) = this->GetParam();
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(Precision::FP32);
+    auto params = ngraph::builder::makeParams(ngPrc, {IS});
+    auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+    auto axisNode = std::make_shared<ngraph::op::Constant>(ngraph::element::Type_t::i64, ngraph::Shape{}, std::vector<int64_t>{-1})->output(0);
+    auto cumSum = std::dynamic_pointer_cast<ngraph::opset4::CumSum>(ngraph::builder::makeCumSum(paramOuts[0], axisNode, false, false));
```

Comment:
Minor comment, you can use ready subgraph from [`ngraph::builder::subgraph`](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/ngraph_functions/include/ngraph_functions/subgraph_builders.hpp)

Reply:
can I just keep it as is?

Reply:
Yes, sure. I do not insist to use it. Just notify you. I think it's more useful